frontend/pages/hosts/ManageHostsPage/ManageHostsPage.jsx

```diff
@@ -147,24 +149,48 @@ export class ManageHostsPage extends PureComponent {
       isAllMatchingHostsSelected: false,
       searchQuery: "",
       hosts: [],
-      isHostsLoading: false,
-      isTeamsLoading: true,
-      sortBy: [],
+      isHostsLoading: true,
+      sortBy: [{ id: DEFAULT_SORT_HEADER, direction: DEFAULT_SORT_DIRECTION }],
+      isConfigLoaded: null,
+      isTeamsLoaded: null,
+      isBasicTier: null,
     };
   }
 
   componentDidMount() {
-    const { dispatch, isBasicTier } = this.props;
+    const { dispatch } = this.props;
     dispatch(getLabels());
-    if (isBasicTier) {
-      dispatch(teamActions.loadAll({}));
-    }
   }
 
-  componentDidUpdate(prevProps) {
-    const { dispatch, isBasicTier } = this.props;
-    if (isBasicTier !== prevProps.isBasicTier && isBasicTier) {
-      dispatch(teamActions.loadAll({}));
+  componentWillReceiveProps() {
+    const { config, dispatch } = this.props;
+    const { isConfigLoaded } = this.state;
+
+    if (!isConfigLoaded && !isEmpty(config)) {
+      const isBasicTier = permissionUtils.isBasicTier(config);
+      if (isBasicTier) {
+        dispatch(teamActions.loadAll({}))
+          .then(() => {
+            this.setState({
+              isBasicTier,
+              isTeamsLoaded: true,
```

Comment:
What if we just don't track `isTeamsLoaded`? I think the UX might be basically the same but the code a bit simpler.

Reply:
My idea here was to use this state to conditionally render a placeholder for the teams dropdown (line 731). Without that check, I was seeing some undesirable UX in edge cases.